src/main/api/opds.ts

```diff
@@ -5,31 +5,41 @@
 // that can be found in the LICENSE file exposed on Github (readium) in the project repository.
 // ==LICENSE-END==
 
-import { inject, injectable} from "inversify";
-
+import * as debug_ from "debug";
+import { inject, injectable } from "inversify";
+import { httpGet } from "readium-desktop/common/utils/http";
+import { OpdsFeedView, THttpGetOpdsResultView } from "readium-desktop/common/views/opds";
+import { OpdsFeedViewConverter } from "readium-desktop/main/converter/opds";
+import { OpdsFeedRepository } from "readium-desktop/main/db/repository/opds";
 import { JSON as TAJSON } from "ta-json-x";
-
 import * as xmldom from "xmldom";
 
-import {
-    convertOpds1ToOpds2,
-} from "@r2-opds-js/opds/converter";
-
+import { convertOpds1ToOpds2 } from "@r2-opds-js/opds/converter";
 import { OPDS } from "@r2-opds-js/opds/opds1/opds";
-
 import { OPDSFeed } from "@r2-opds-js/opds/opds2/opds2";
 import { XML } from "@r2-utils-js/_utils/xml-js-mapper";
 
-import { OpdsFeedView, THttpGetOpdsResultView } from "readium-desktop/common/views/opds";
-
-import { OpdsFeedViewConverter } from "readium-desktop/main/converter/opds";
-
-import { OpdsFeedRepository } from "readium-desktop/main/db/repository/opds";
-
-import { httpGet } from "readium-desktop/common/utils/http";
+// Logger
+const debug = debug_("readium-desktop:src/main/api/opds");
 
 @injectable()
 export class OpdsApi {
+
+    /**
+     * test all possible content-type for both xml and json
+     * @param contentType content-type headers
+     * @returns if content-Type is missing accept
+     */
+    public static contentTypeisAccepted(contentType: string) {
```

Comment:
`contentType` can be undefined, so the parameter's TypeScript type should be `string | undefined`.
Digging deeper: `contentType` is an optional field in `IHttpGetResult<TBody, TData>`, typically `undefined` when `response.caseless.get(...)` does not match anything.
Test case: `node -e "const url = 'https://flaschenpost.piratenpartei.de/catalog/'; require('request')(url, (err, res) => { console.log(res.headers); console.log(res.caseless.get('Content-Type')); });"`

Reply:
yes i fixed the type https://github.com/readium/readium-desktop/pull/542/commits/eb40109765dec0a3dc4092d277e3c7c42e4770e8